Solve column equation and add distributed loads

lfverlaan opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Fixes #28065

#### Brief description of what is fixed or changed
Firstly, support has been added for distributed loads. Also loads (including distributed loads) can now be removed. The unknown reaction forces can now also be solved.

#### Other comments
This pull request builds upon pull request #28049, see the changes in lfverlaan#4.

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* physics.continuum_mechanics
  * Added a new `Column` class for modeling axially loaded structural elements, similar in structure to the `Beam` class.
  * Supports applying axial point loads and fixed horizontal supports using singularity functions.
  * Implemented support for distributed axial loads (with optional `end` location) via singularity function expansion.
  * Added functionality to remove previously applied loads (`remove_load`), updating the internal load equation accordingly.
  * Automatically assembles the total axial load equation symbolically using singularity functions.
  * Introduced a solver method (`solve_for_reaction_loads`) to compute unknown reaction forces by applying equilibrium and boundary conditions.

<!-- END RELEASE NOTES -->

lfverlaan committed:
New files

Continuum mechanics: New files

Two new files have been added, column.py and a test file for the column script.

lfverlaan committed:
Update .mailmap

lfverlaan committed:
Update test_column.py

Tests have been written for the initialization of the column module, and the formulation of the load equation.

lfverlaan committed:
Update test_column.py

Adjusted the tests. Added a test to show that adding three supports is also compatible. Also added a test to check if reaction forces due to supports are also added to the load equation. Also removed some tests, inputs that are outside of the scope.

lfverlaan committed:
Update column.py

The base of the Column class has been written. A column with a certain length, area and elastic modulus can be initialized. Methods have also been added to apply supports and forces. These methods add the singularity functions to the load equations. The user can get the load equation through the 'load' property.

Tom-van-Woudenberg committed:
Add docs

lfverlaan committed:
Update column.py

A little bit of code cleanliness, removed whitespace

lfverlaan committed:
Merge pull request #1 from Tom-van-Woudenberg/docs

Add docs

lfverlaan committed:
Update column.py

Fixed a mistake where the reaction forces got added to the applied loads list. Now they are simply added to the load equation instead of the apply_load() function being called.

Also added examples in the docstring of the functions.

lfverlaan committed:
Update column.py

Checked the spelling, neatness.

Tom-van-Woudenberg committed:
Fix error in docs

Fixes Title overline too short

lfverlaan committed:
Merge pull request #2 from Tom-van-Woudenberg/docs

Fix error in docs

lfverlaan committed:
Update column.py

removed trailing whitespace

lfverlaan commented:
I have made a new issue and pull request to work on the next set of problems. Is it fine to work like this, given the pull request for the first part has not been merged yet?

@Tom-van-Woudenberg

lfverlaan committed:
Update test_column.py

Added tests for the remove load function and the addition of higher order loads.

Tom-van-Woudenberg commented:
> I have made a new issue and pull request to work on the next set of problems. Is it fine to work like this, given the pull request for the first part has not been merged yet?

I think it is. Worst case scenario is that you've to make changes in the other pull request which are not compatible with your approach in the added features in this pull request. But you can assess the risk of that yourself best.

Tom-van-Woudenberg commented:
@saiudayagiri, any insights from your side here?

lfverlaan committed:
Update test_column.py

Adjusted the tests of the test_column to have an input for new variable 'end'. Also fixed some incorrect load equestions

saiudayagiri commented:
@Tom-van-Woudenberg, @lfverlaan, I agree #28066 may need rework if #28049’s changes (e.g., join test) conflict, as Tom noted. Tom, as you know better, doesn’t #28066’s diff showing #28049’s lines as new (post-merge without rebase) cause confusion and extra review effort? An alternate solution could be stacking PRs by basing #28066 on #28049’s branch, updating post-merge, to isolate changes and ease review.

lfverlaan committed:
test_column

Added tests for the solve_for_reaction_loads function. There are tests for statically determnate and indeterminate columns.

lfverlaan committed:
Update column.py

Methods for removing a load and solving the reaction forces have been constructed. Distributed loads have also been integrated, using a helper function called _taylor_helper().

lfverlaan commented:
![image](https://github.com/user-attachments/assets/41ee3e79-9368-44d5-91cd-50d73440c66a)
When comparing the branches, git wants me to start a new pull request. Is this what you mean @Tom-van-Woudenberg?

Tom-van-Woudenberg commented:
>  When comparing the branches, git wants me to start a new pull request. Is this what you mean?

We discussed this in-person right? YOu could create that pull request in your personal repo and link to it just to visualize the diff
